packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -25,7 +25,16 @@ import { logError } from '../shared/logger';
 import { invokeEventListener } from './invoker';
 import { getVMBeingRendered } from './template';
 import { EmptyArray, EmptyObject } from './utils';
-import { getAssociatedVM, runConnectedCallback, SlotSet, VM, VMState } from './vm';
+import {
+    runConnectedCallback,
+    SlotSet,
+    VM,
+    VMState,
+    getAssociatedVMIfPresent,
+    removeVM,
+    rerenderVM,
+    appendVM,
+} from './vm';
```

Comment:
hooks.ts should not have access to any VM related operation, should be very generic about the diffing algo operations. This refactor is moving that logic back to api.ts for that matter. Eventually, most of these operations here will also be moved out, but that's another refactor to be made as part of the node-reactions